app/models/watcher_notification_mailer.rb

```diff
@@ -0,0 +1,48 @@
+#-- encoding: UTF-8
+#-- copyright
+# OpenProject is a project management system.
+# Copyright (C) 2012-2015 the OpenProject Foundation (OPF)
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See doc/COPYRIGHT.rdoc for more details.
+#++
+
+class WatcherNotificationMailer
+  class << self
+    def handle_watcher(watcher_id, watcher_setter_id)
+      unless other_jobs_queued?(Watcher.find(watcher_id).watchable)
+        job = DeliverWatcherNotificationJob.new(watcher_id, watcher_setter_id)
+        Delayed::Job.enqueue job
+      end
+    end
+
+    private
+
+    # HACK: TODO this needs generalization as well as performance improvements
+    # We need to make sure no work package created or updated job is queued to avoid sending two
+    # mails in short succession.
+    def other_jobs_queued?(work_package)
+      Delayed::Job.where('handler LIKE ?', "%journal_id: #{work_package.journals.last.id}%").exists?
```

Comment:
Can you eliminate the possibility of some unrelated job including `journal_id: #{id}` and thus causing a false-positive here? I'd probably prefer If we could identify all related jobs for this journal, but this may be unfeasible.